app/src/main/java/com/example/trainingintive/util/BindingAdapters.kt

```diff
@@ -12,3 +15,11 @@ fun bindImage(imgView: ImageView, imgUrl: String?) {
         imgView.load(imgUri)
     }
 }
+
+@BindingAdapter("data")
+fun bindDataToRecyclerView(recyclerView: RecyclerView, items: List<ActivityModel>) {
+    val adapter = recyclerView.adapter
+    if (adapter is ActivityModelAdapter) {
```

Comment:
For now, it's okay because you are using only one adapter but what will you do layer when for e.g. you will have 10 RecyclerViews with 10 adapters? This `BindingAdapter` should be generic. You can leave a TODO for this and implement it later (to achieve this you will probably need some generic adapter class).

Reply:
My idea for later was set items parameter as List<Any> and then in "when" block cast to appropriate list, is it good way?

Reply:
For sure it is a better way instead of creating a new `BindingAdapter` for each type of data but I would not say that it is a good way :). The cleaner way would be to create some base (generic) class for the Adapter with some generic function to update the data. Then, in your `BindingAdapter` you will simply call some function, this way the `BindingAdapter` will not need to know the type.